ego4d/internal/university_files.py

```diff
@@ -0,0 +1,721 @@
+                    f" video metadata'\n"
+                )
+
+
+
+    # Check ids in annotations are in video_metadata
+    if annotations_dict:
+        for video_id in annotations_dict:
+            if video_id not in video_metadata_dict:
+                error_message += (
+                    "annotations points to video_metadata at"
+                    f" non-existent university_video_id: '{video_id}'\n"
+                )
+
+    # Check from video_metadata:
+    video_ids = set()
+    for video_id, video_metadata in video_metadata_dict.items():
+        #   0. check no duplicate university_video_id
+        if video_id in video_ids:
+             error_message += (
+                    f"Video: {video_id} has duplicate video id \n"
+                )
+        video_ids.add(video_id)
+        #   1. participant_ids are in participant_dict
+        if participant_dict:
+            if video_metadata.recording_participant_id not in participant_dict:
+                error_message += (
+                    f"Video: {video_id} points to participant file at"
+                    " non-existent participant_id: "
+                    f"'{video_metadata.recording_participant_id}'\n"
+                )
+            if video_metadata.recording_participant_id is None:
+                error_message += (
+                    f"Video: {video_id} does not have corresponding"
+                    " participant_id \n"
+                )
+
+        #   2. scenario_ids are in scenarios
+        for scenario_id in video_metadata.video_scenario_ids:
+            if scenario_id not in scenarios:
+                error_message += (
+                    f"Video: {video_id} points to scenarios at"
+                    f" non-existent scenario_id: '{scenario_id}'\n"
+                )
+
+        #   3. device_ids are in devices
+        if video_metadata.device_id and video_metadata.device_id not in devices:
+            error_message += (
+                f"Video: {video_id} points to devices at"
+                f" non-existent device_id: '{video_metadata.device_id}'\n"
+            )
+
+        #   4. physical_settings are in physical_setting
+        if physical_setting_dict:
+            if (
+                video_metadata.physical_setting_id
+                and video_metadata.physical_setting_id not in physical_setting_dict
+            ):
+                error_message += (
+                    f"Video: {video_id} points to physical setting at"
+                    " non-existent physical_setting_id: "
+                    f"'{video_metadata.physical_setting_id}'\n"
+                )
+
+        #   5. university_video_ids are in components
+        if video_id not in video_components_dict:
+            error_message += (
+                f"Video: {video_id} points to component at"
+                " non-existent university_video_id: "
+                f"'{video_id}'\n"
+            )
+
+    if error_message:
+        print(error_message)
```

Comment:
Let's return the error messages and save them to disk as well as printing them to stdout